platform/macos/src/MGLMapView.mm

```diff
@@ -1695,6 +1696,21 @@ - (void)setRotateEnabled:(BOOL)rotateEnabled {
     _compass.hidden = !rotateEnabled;
 }
 
+- (void)adjustZoomLevelForKey:(NSString *)key {
+    static NSDictionary<NSString *, void(^)(void)> *zoomOpDict = nil;
+
+    if (zoomOpDict == nil) {
+        void (^zoomOut)(void) = ^ { [self setZoomLevel:(self.zoomLevel - 1) animated:YES]; };
+        void (^zoomIn)(void) = ^ { [self setZoomLevel:(self.zoomLevel + 1) animated:YES]; };
+        zoomOpDict = @{@"-": zoomOut, @"+": zoomIn, @"=": zoomIn};
```

Comment:
True, so you could say `[key characterAtIndex:0]` and switch over that `unichar`.